### PR TITLE
Fix #18314 - dragged row in index form

### LINKED ISSUE #18314: Issue reordering columns in index form
### Describe the bug

In the index form when reorderring columns by dragging the width of the dragged row becomes very  small.

### To Reproduce

1. Go to Structure / Create Index
2. In the form start dragging the index column

### Expected behavior

The form inputs should remain visible during the drag duration

### Screenshots

image
### Server configuration

 e808706 on https://demo.phpmyadmin.net/

### Client configuration

- Browser: Firefox 111.0
- Operating system: Linux

### PR DESCRIPTION
**Pull request by IAndreyI**
### Description

This PR fixes a issue with the width of the dragged row in index form. Before this the width was very small. Now the width of the dragged row will be the same as the original.
The problem was in the lines below. It doesn’t work here. 
```js
$indexColumns.find('td').each(function () {
    $(this).css('width', $(this).width() + 'px');
});
```

Before

https://user-images.githubusercontent.com/36403726/236654450-01f783e9-4040-4f9f-8202-39c963afe622.mp4

After

https://user-images.githubusercontent.com/36403726/236654476-a61e2271-cdbe-4fac-a70c-120b32724554.mp4

Fixes #18314

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

File: js/src/functions.js [hunk 1]
```diff
@@ -3518,13 +3518,27 @@ Functions.showIndexEditDialog = function ($outer) {
     Indexes.checkIndexType();
     Functions.checkIndexName('index_frm');
     var $indexColumns = $('#index_columns');
-    $indexColumns.find('td').each(function () {
-        $(this).css('width', $(this).width() + 'px');
-    });
     $indexColumns.find('tbody').sortable({
```
Comment (williamdes) [inline js/src/functions.js:3521]: did you check if there is other `.sortable` that could be affected ?
to know if the helper function should be extracted into a global function

File: js/src/functions.js [hunk 2]
```diff
@@ -4681,4 +4695,4 @@ Functions.getPostData = function () {
     }
     return dataPost;
 };
-jQuery.fn.getPostData = Functions.getPostData;
+jQuery.fn.getPostData = Functions.getPostData;
```
Comment (williamdes) [inline js/src/functions.js:4698]: looks like you may want to install a plugin like editorconfig for vscode
anyway, that's okay I can add it back

File: js/src/functions.js [hunk 3]
```diff
@@ -3518,13 +3518,27 @@ Functions.showIndexEditDialog = function ($outer) {
...
+            var $originalCells = tr.children();
+            var $helper = tr.clone();
+            $helper.children().each(function (index) {
+                // Set cell width in dragged row
+                $(this).width($originalCells.eq(index).outerWidth());
+                var $childrenSelect = $originalCells.eq(index).children('select');
+                if ($childrenSelect.length) {
+                    var selectedIndex = $childrenSelect.prop('selectedIndex');
+                    // Set correct select value in dragged row
+                    $(this).children('select').prop('selectedIndex', parseInt(selectedIndex));
```
Comment (williamdes) [inline js/src/functions.js:3537]:
```suggestion
                    $(this).children('select').prop('selectedIndex', parseInt(selectedIndex, '10'));
```

Comment (williamdes) [reply js/src/functions.js:3537]: Is parseint necessary ?

**Review by williamdes — APPROVED** [review]
Awesome contribution, 💯 for the videos and everything else 👏🏻

File: js/src/functions.js (hunk 1 above)
Comment (IAndreyI) [reply js/src/functions.js:3521]: I also found `.sortable` in indexes.js and routines.js. Can I create a global function "makeTableSortable" in functions.js?

File: js/src/functions.js (hunk 2 above)
Comment (IAndreyI) [reply js/src/functions.js:4698]: I removed this line because I got a message about this problem
изображение

File: js/src/functions.js (hunk 3 above)
Comment (IAndreyI) [reply js/src/functions.js:3537]: No. I'll remove it

File: js/src/functions.js (hunk 2 above)
Comment (williamdes) [reply js/src/functions.js:4698]: Hmm okay, but `insert_final_newline = true` is different from whitespaces.
Best if to revert this line change and if the bot is wrong we will fix it ^^

See: `.editorconfig`

File: js/src/functions.js (hunk 1 above)
Comment (williamdes) [reply js/src/functions.js:3521]: That could be nice, but only if you can reproduce the bug on the indexes and routines

Comment (IAndreyI) [reply js/src/functions.js:3521]: The bug doesn't reproduce in routines. How can I check this bug in indexes? I don't understand where the indexes.js uses

**Review by MauricioFauth — APPROVED** [review]

**Pull request merged**